File: src/0070-Climbing-Stairs/0070.py
class Solution:
    def _climbStairs(self, n, climb):
        if n == 0 or n == 1:
            return 1

        if n not in climb:
            climb[n] = self._climbStairs(n - 1, climb) + self._climbStairs(n - 2, climb)

        return climb[n]

    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        climb = dict()
        return self._climbStairs(n, climb)

if __name__ == '__main__':
    n = 3
    print(Solution().climbStairs(n))

File: src/0070-Climbing-Stairs/0070.cpp
class Solution {
 public:
  int climbStairs(int n) {
    long long a = 1, b = 1, c;
    for (int i = 0; i < n; i++) {
      c = a;
      a = b;
      b += c;
    }
    return a;
  }
};